app/src/lib/utils/misc.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Throttles the execution of a function.
+ * @param fn The function to be throttled.
+ * @param wait The time to wait between function invocations in milliseconds.
+ * @returns A throttled version of the original function.
```

Comment:
> Type params are redundant with typescript
>

Do you mean the `<T extends (...args: any[]) => any>`? I would disagree.
In this case it is accomplishing 3 things:
1. Narrows down the `fn` argument to be any kind of function
2. Ensures that the return type is always the exact same type of function as the one passed in as an argument.
3. Allows us to keep type-guards when accessing the parameters or return types, dynamically (see lines 12 and 14). 

Is that what you meant, or were you referring to something else?

> and perhaps we can omit the comment too since it's a simple function?
>

Will remove that 👍

Reply:
Sorry, no I meant the jsdoc, not the actual parameter!

Reply:
Ah gotcha :)